starwars/starwars_people/views.py

```diff
@@ -2,24 +2,36 @@
 from rest_framework.response import Response
 from rest_framework.viewsets import ModelViewSet
 from starwars_people.models import Dataset
-from rest_framework import status
+from rest_framework import status, views
 from starwars_people.serializers import DatasetSerializer
-from starwars_people.services import make_pagination
-from starwars_people.tasks import download_dataset
+from starwars_people.services import make_pagination, start_download_dataset_task
+from django.core.cache import cache
 
 
 class DatasetDetailView(ModelViewSet):
     queryset = Dataset.objects.all()
     serializer_class = DatasetSerializer
 
     @action(detail=False, methods=['post'])
-    def import_starwars_data(self, request):
-        download_dataset()
-        return Response(status=status.HTTP_201_CREATED)
+    def download_dataset(self, request) -> Response:
+        cache_task_key = start_download_dataset_task()
```

Comment:
i think you shoud generate cache task key, and then pass it as argument to your `start_download_dataset_task` function